### DISCUSSION TITLE
Log progress bar output

### DISCUSSION THREAD

**Discussion by JoseMMerida** [futureverse/progressr#145] [Q&A]
Hi,

I am trying to log the progress bar in a dockerized task to be able to see and potentially debug when I encounter any problem in one of my parallelized tasks.

While I have the enviroment variable `R_PROGRESSR_ENABLE=TRUE` I am not able to log the bar to the log output.

Just FYI I am using kubernetes to deploy my tasks. 

Does anybody has any hint on how to proceed from here?

**Comment by HenrikBengtsson** [comment]
Hi. I'm not 100% of your setup, but the default is for all **progressr** handlers that output to the terminal to output to the standard error (stderr). This is by design.

Given this, is there are reason why you cannot redirect also stderr to your log file, e.g.

```r
Rscript my_script.R &> my_script.out
```

Alternatively, you can adjust the defaults of the [txtprogressbar handler](https://progressr.futureverse.org/reference/handler_txtprogressbar.html) arguments so that it outputs to stdout instead of the default stderr:

```r
progressr::handlers(handler_txtprogressbar(file = stdout()))
```

**Comment by JoseMMerida** [comment]
Thanks for your quick answer!

I am executing an Rscript through a kubernetes job, while trying to capture the output via promtail.

It is kind of wierd, because the typical outputs streamed to stderr/stdout I am able to see them. But it is a different story when trying to capture the progress bar.

While attaching to the pod and executing the command, I am also able to see the output of the progress bar to the console.

I am using the progress handler to build my progress bar.
